### PR TITLE
[FLINK-34688][cdc-connector] Fix: The assigner is not ready to offer finished split information, this should not be called

### PR DESCRIPTION
**Pull request by loserwang1024**
As described in https://issues.apache.org/jira/browse/FLINK-35859 , only discover new added table when `assignerStatus == INITIAL_ASSIGNING_FINISHED or || assignerStatus == NEWLY_ADDED_ASSIGNING_FINISHED`. It also same with what java doc says:

image

### CODE REVIEW

**Comment by leonardBang** [comment]
Thanks for fix this ticket, @loserwang1024 Would you like to add a test?

File: flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mysql-cdc/src/test/java/org/apache/flink/cdc/connectors/mysql/source/assigners/MySqlSnapshotSplitAssignerTest.java [hunk 1]
```diff
@@ -484,15 +484,33 @@ public void testEnumerateTablesLazily() {
...
-        assertEquals(expected, getTestAssignSnapshotSplitsFromCheckpoint());
+        assertEquals(
+                expected,
+                getTestAssignSnapshotSplitsFromCheckpoint(
+                        AssignerStatus.INITIAL_ASSIGNING_FINISHED));
+    }
+
+    @Test
+    public void
+            testScanNewlyAddedTableStartFromCheckpoint_NEWLY_ADDED_ASSIGNING_SNAPSHOT_FINISHED() {
```
Comment (leonardBang) [inline flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mysql-cdc/src/test/java/org/apache/flink/cdc/connectors/mysql/source/assigners/MySqlSnapshotSplitAssignerTest.java:503]: minor: The method name should use camel case

File: flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mysql-cdc/src/test/java/org/apache/flink/cdc/connectors/mysql/source/assigners/MySqlSnapshotSplitAssignerTest.java [hunk 2]
```diff
@@ -536,7 +554,7 @@ private List<String> getTestAssignSnapshotSplits(
         return getSplitsFromAssigner(assigner);
     }
 
-    private List<String> getTestAssignSnapshotSplitsFromCheckpoint() {
+    private List<String> getTestAssignSnapshotSplitsFromCheckpoint(AssignerStatus assignerStatus) {
```
Comment (leonardBang) [inline flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mysql-cdc/src/test/java/org/apache/flink/cdc/connectors/mysql/source/assigners/MySqlSnapshotSplitAssignerTest.java:557]: minor: The method name should use camel case

Comment (loserwang1024) [reply flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mysql-cdc/src/test/java/org/apache/flink/cdc/connectors/mysql/source/assigners/MySqlSnapshotSplitAssignerTest.java:556]: done

File: flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mysql-cdc/src/test/java/org/apache/flink/cdc/connectors/mysql/source/assigners/MySqlSnapshotSplitAssignerTest.java (hunk 1 above)
Comment (loserwang1024) [reply flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mysql-cdc/src/test/java/org/apache/flink/cdc/connectors/mysql/source/assigners/MySqlSnapshotSplitAssignerTest.java]: done

**Review by leonardBang — APPROVED** [review]
Thanks @loserwang1024  for the update, +1

**Pull request merged**